underscore.js

```diff
@@ -1009,11 +1009,21 @@
 
   // Convert an object into a list of `[key, value]` pairs.
   _.pairs = function(obj) {
-    var keys = _.keys(obj);
-    var length = keys.length;
-    var pairs = Array(length);
-    for (var i = 0; i < length; i++) {
-      pairs[i] = [keys[i], obj[keys[i]]];
+    var pairs, i, length;
+    if (_.isMap(obj)) {
+      var entries = obj.entries();
```

Comment:
You can use `Map#forEach` for all environments that support `Map`. You can then recycle the same code for `_.toPairs` to enable `_.toArray` to convert maps to entries too. I've found lib methods like this useful to support various Node versions without worrying about shims or what narrow support paths are available.